src/main/java/net/fabricmc/loom/api/fabricapi/DataGenerationSettings.java

```
@@ -67,4 +72,15 @@ public interface DataGenerationSettings {
 	 * Contains a boolean property indicating whether data generation will be compiled and ran with the client.
 	 */
 	Property<Boolean> getClient();
+
+	/**
+	 * Sets {@link #getModId()} property based on the {@code id} field defined in the provided file.
+	 */
+	default void modId(File fabricModJsonFile) {
+        try {
+            getModId().set(Objects.requireNonNull(FabricModJsonFactory.createFromOverrideNullable(fabricModJsonFile)).getId());
```

This should fail if its an invalid FMJ.